hawkular-inventory-json-helper/src/main/java/org/hawkular/inventory/json/mixins/DataEntityMixin.java

```diff
@@ -19,20 +19,26 @@
 import java.util.Map;
 
 import org.hawkular.inventory.api.model.CanonicalPath;
+import org.hawkular.inventory.api.model.DataEntity;
 import org.hawkular.inventory.api.model.StructuredData;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * @author Lukas Krejci
  * @since 0.3.0
  */
-public class DataEntityMixin {
+public abstract class DataEntityMixin {
 
     @JsonCreator
     public DataEntityMixin(@JsonProperty("path") CanonicalPath path, @JsonProperty("value") StructuredData value,
             @JsonProperty("properties") Map<String, Object> properties) {
     }
 
+    @JsonIgnore abstract String getId();
```

Comment:
btw. I think, we should test the form of the json, currently what we do is checking if 
```haskell
deserialization . serialization ~ id
```
That's completely ok, but it doesn't say a word about the JSON format